libs/template/presets/storefront/experience/header.ts

```diff
@@ -93,19 +93,61 @@ export const HeaderTemplate: StaticComponent = {
           type: 'oryx-composition',
           components: [
             {
-              type: 'oryx-site-navigation-item',
-              options: {
-                data: {
-                  contentBehavior: 'dropdown',
-                  label: 'USER.NAME',
-                  icon: IconTypes.User,
+              type: 'oryx-composition',
+              visibility: {
+                token: 'USER.AUTHENTICATED',
+              },
+              components: [
+                {
+                  type: 'oryx-site-navigation-item',
+                  options: {
+                    data: {
+                      label: 'login',
+                      icon: IconTypes.User,
+                      url: { type: 'login' },
+                    },
+                  },
                 },
+              ],
+            },
+            {
+              type: 'oryx-composition',
+              visibility: {
+                token: 'USER.!AUTHENTICATED',
               },
-              components: [{ type: 'oryx-auth-login-link' }],
+              components: [
+                {
+                  type: 'oryx-site-navigation-item',
+                  options: {
+                    data: {
+                      contentBehavior: 'dropdown',
+                      label: 'USER.NAME',
+                      icon: IconTypes.User,
+                    },
+                  },
+                  components: [{ type: 'oryx-auth-login-link' }],
+                },
+              ],
+            },
+            {
+              type: 'oryx-composition',
+              visibility: {
+                hide: false,
+                token: 'CART.!EMPTY',
+              },
```

Comment:
We like visibility to be part of the options model, as the idea has always been to not have too many data fields in the component model.

Reply:
We discussed it with Stan, and considered that visibility config should have even separate tab in EB components configuration, and more logical to keep them separate. Agree, this is debatable and we can discuss it again

Reply:
yes it might have a separate tab, similar to layout and styling, but I would still keep it in options (similar to layout and styling btw)

Reply:
I'm not against this, but i'm a little worried that we are polluting options with reserved fields. We already have rules for styling, and plan to add visibility there, this can cause potential issues, if some component have its own options with same names

Reply:
Yes, that's a valid concern. I had a conversation with Stan on this topic and we came out with the following conclusions:

- we like to make "visibility" part of the option rules. The option rules have a `query` property that allows to put conditions for a rule.  With this mechanism we can currently apply different style rules per screen size (aka breakpoint), or styles for the child components. 
- the visibility is one of the outcomes of the conditions that you're build. The visibility logic must remain in the composition component to ensure that the component is not rendered inside the DOM
- going forward, we probably want to move the query up, so that options can also become conditional. We will do this in follow up work
- We do not expect to support screen size driven conditions yet, but that's also one of the things next up. If we combine this with the previous point, than we can probably move the `ResizeObserver` from layout to composition, so that all options (including layout) are filtered out up front. This would simplify the stuff nicely, and make it more efficient as well. 

So, for layout we currently have this model:

```ts
const options = {
  data: {
    rules: [
      {
        layout: "carousel",
        padding: "30px 0 5px",
        align: "stretch",
      },
      {
        query: {
          breakpoint: "sm",
        },
        padding: "20px",
      },
    ],
    category: "10",
    sort: "rating",
  },
};
```

We like to move visibility (conditions) into this model:

```ts
const options = {
  data: {
    rules: [
      {
        layout: "carousel",
        padding: "30px 0 5px",
        align: "stretch",
      },
      {
        query: {
          condition: 'USER.AUTHENTICATED',
        },
        hidden: true
      },
    ],
    category: "10",
    sort: "rating",
  },
};
```

Reply:
Created a follow up ticked based on your suggestion